src/content/reference/react/cache.md

```diff
@@ -0,0 +1,306 @@
+---
+title: cache
+canary: true
+---
+
+<Canary>
+`cache` is currently only available in React’s [canary](https://react.dev/community/versioning-policy#canary-channel) and [experimental](https://react.dev/community/versioning-policy#experimental-channel) channels. Please ensure you understand the limitations before using `cache` in production. 
+
+Learn more about [React's release channels here](/community/versioning-policy#all-release-channels). 
+</Canary>
+
+<Intro>
+
+`cache` lets you cache the result of a data fetch or computation.
+
+```js
+const cachingFn = cache(fn);
```

Comment:
Yea I've struggled with naming on this. What would be the difference for _does_ vs. _with_? The function returned will cache the result given the input. Is that not a caching function? I'm not strongly opposed to say "function _with_ caching" but want to build up a strong mental model for it

Reply:
> What would be the difference for does vs. with?

The `cache` function _does_ the caching, and returns an instance of the provided function with the caching that `cache` provides. When I read `cachingFn`, I read it as "a function that can cache something" which makes me think that I will use that function to then cache something, something like:

```js
const cachingFn = cache(fn);

// Wrong
const cachedFn = cachingFn(otherFn);
```

I think it's because caching is a verb, so I read it as a function that will do caching, instead of a function that has had caching done to it. Similar to memoize, the returned value is not a memoizing or caching function, it's a memoized or cached function.